src/Microsoft.AspNetCore.SystemWebAdapters.FrameworkServices/ProxyHeaderModule.cs

```diff
@@ -103,6 +110,31 @@ private static void UseForwardedFor(NameValueCollection requestHeaders, NameValu
         }
     }
 
+    private class ReflectionHelper
+    {
+        private readonly FieldInfo _wrField;
+        private readonly Type _iis7WorkerRequestType;
+        private readonly FieldInfo _knownRequestHeadersField;
+
+        public ReflectionHelper()
+        {
+            _wrField = typeof(HttpRequest).GetField("_wr", BindingFlags.NonPublic | BindingFlags.Instance);
+            _iis7WorkerRequestType = Assembly.GetAssembly(typeof(HttpRequest)).GetType("System.Web.Hosting.IIS7WorkerRequest");
+            _knownRequestHeadersField = _iis7WorkerRequestType.GetField("_knownRequestHeaders", BindingFlags.NonPublic | BindingFlags.Instance);
+        }
+        public void SetHostHeader(HttpRequest? request, string host)
```

Comment:
Can we require this to be non-nullable?

Reply:
Its the unit tests that pass in null, I was being lazy and not wanting to change the tests to have to mock some kind of httprequest

Reply:
I made this non nullable and just had the callers null check for the unit tests